sklearn/utils/estimator_checks.py

```diff
@@ -82,10 +82,17 @@
 
 
 def _yield_api_checks(estimator):
+    tags = get_tags(estimator)
+    yield check_estimator_cloneable
     yield check_estimator_repr
     yield check_no_attributes_set_in_init
     yield check_fit_score_takes_y
     yield check_estimators_overwrite_params
+    yield check_dont_overwrite_parameters
+    yield check_estimators_fit_returns_self
+    yield partial(check_estimators_fit_returns_self, readonly_memmap=True)
+    if tags.requires_fit:
+        yield check_estimators_unfitted
```

Comment:
For me this is also part of the API. I think this is good to raise the proper error message. We could make sure that this test mentioned that you should either set the tag or to use the utility `check_is_fitted`.